### PR TITLE
Update Fortran Standard to 2008

### PR DESCRIPTION
**Pull request by akleb**
## Purpose
Update the Fortran standard to 2008. I picked 2008 because this datbase relies on [norm2](https://gcc.gnu.org/onlinedocs/gfortran/NORM2.html), which comes from the 2008 standard.

## Type of change
What types of change is it?

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [X] Maintenance update
- [ ] Other (please describe)

## Testing
I have not tested this with TAPENADE. I also don't think I currently have any access to intel compiler, so I cannot check the intel flags or compilation. (I guess we probably have a docker image I can try?)

## Checklist

- [X] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [X] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Comment by sseraj** [comment]
> I have not tested this with TAPENADE

pyHyp is not differentiated so this not an issue

File: config/defaults/config.LINUX_GFORTRAN_OPENMPI.mk [hunk 1]
```diff
@@ -18,10 +18,10 @@ CGNS_INCLUDE_FLAGS=-I$(CGNS_HOME)/include
 CGNS_LINKER_FLAGS=-L$(CGNS_HOME)/lib -lcgns
 
 # ------- Define Compiler Flags ----------------------------------------
-FF90_GEN_FLAGS = -fPIC
+FF90_GEN_FLAGS = -fPIC -std=f2008
 CC_GEN_FLAGS   = -fPIC
 
-FF90_OPT_FLAGS   =  -fPIC -fdefault-real-8 -O2
+FF90_OPT_FLAGS   =  -fPIC -fdefault-real-8 -O2 -std=f2008
```
Comment (nwu63) [inline config/defaults/config.LINUX_GFORTRAN_OPENMPI.mk:24]: I don't think you need to add this to both, and in fact `-fPIC` probably doesn't need to be added to both. The actual flags are just combined I believe -- see [here](https://github.com/akleb/pyhyp/blob/std-f2008/Hyp_Common.mk#L51).

File: config/defaults/config.LINUX_INTEL_OPENMPI.mk [hunk 2]
```diff
@@ -17,10 +17,10 @@ CGNS_INCLUDE_FLAGS=-I$(CGNS_HOME)/include
 CGNS_LINKER_FLAGS=-L$(CGNS_HOME)/lib -lcgns
 
 # ------- Define Compiler Flags ----------------------------------------
-FF90_GEN_FLAGS = -fPIC
+FF90_GEN_FLAGS = -fPIC -stand f08
```
Comment (nwu63) [inline config/defaults/config.LINUX_INTEL_OPENMPI.mk:20]: Same here, but good call for `-stand` I'll do the same for my PRs.

File: config/defaults/config.LINUX_GFORTRAN_OPENMPI.mk (hunk 1 above)
Comment (akleb) [reply config/defaults/config.LINUX_GFORTRAN_OPENMPI.mk]: Yeah, that `-fPIC` confused me.

**Review by nwu63 — APPROVED** [review]

**Review by sseraj — COMMENTED** [review]
The changes all look reasonable. Can you take a quick look at the Intel build log on Azure and see if any of the warnings are unexpected?

**Comment by nwu63** [comment]
Looks fine to me, they are the same stuff as discussed in mdolab/adflow#53.

**Review by sseraj — APPROVED** [review]
I see, they are unrelated to the standard.

**Pull request merged**